<launch>
<arg name="robot_base" default="0.194" />

<node name="arduino" pkg="rosserial_arduino" type="serial_node.py" >
    <param name="port" type="string" value="/dev/ttyACM0"/>
    <param name="baud" type="int" value="57600"/>
    <param name="robot_base_L" type="double" value="$(arg robot_base)"/>

</node>

  <!-- RPLidar-->
  <node name="rplidarNode"          pkg="rplidar_ros"  type="rplidarNode" output="screen">
  <param name="serial_port"         type="string" value="/dev/ttyUSB0"/>
  <param name="serial_baudrate"     type="int"    value="115200"/><!--A1/A2 -->
  <param name="frame_id"            type="string" value="laser"/>
  <param name="inverted"            type="bool"   value="false"/>
  <param name="angle_compensate"    type="bool"   value="true"/>
  </node>

  <!-- wheels encoders odometry-->
  <remap from="wheel_encoders/odom" to="odom" />
  <node name="EncodersOdometry" pkg="siryo" type="EncodersOdometry" >
    <param name="robot_base_L" type="double" value="$(arg robot_base)"/>
    <param name="publish_tf" type="bool" value="true"/>
  </node>

  <include file="$(find siryo)/launch/camera.launch" />


  <node pkg="tf" type="static_transform_publisher" name="base_link_to_laser" args="0.03 0.0 0.13 1.571 0 0 base_link laser 100" />
</launch>


